RULES/hdl/questa.go

```diff
@@ -2,75 +2,569 @@ package hdl
+			}
+		} else if strings.HasPrefix(arg, "-verbosity=") {
+			// Define verbosity level
+			var level string
+			if _, err := fmt.Sscanf(arg, "-verbosity=%s", &level); err == nil {
+				verbosity_flag, print_output = verbosityLevelToFlag(level)
+			} else {
+				log.Fatal("-verbosity expects an argument of 'low', 'medium', 'high' or 'none'!")
+			}
+		} else if strings.HasPrefix(arg, "+") {
+			// All '+' arguments go directly to the simulator
+			plusargs_flag = plusargs_flag + " " + arg
+		}
+	}
+
+	// Create optional command preamble
+	cmd_preamble, testcase = preamble(rule, testcase)
+
+	cmd_echo := ""
+	if rule.Params != nil && params != "" {
+		// Update coverage database name based on parameters. We cannot merge
+		// different parameter sets, do we have to make a dedicated main database
+		// for this parameter set.
+		main_coverage_db = main_coverage_db + "_" + params
+		coverage_db = coverage_db + "_" + params
+		cmd_echo = "Testcase " + params
+
+		// Update with testcase if specified
+		if testcase != "" {
+			coverage_db = coverage_db + "_" + testcase
+			cmd_echo = cmd_echo + "/" + testcase + ":"
+			testcase = params + "_" + testcase
+		} else {
+			cmd_echo = cmd_echo + ":"
+			testcase = params
+		}
+	} else {
+		// Update coverage database name with testcase alone, main database stays
+		// the same
+		if testcase != "" {
+			coverage_db = coverage_db + "_" + testcase
+			cmd_echo = "Testcase " + testcase + ":"
+		} else {
+			testcase = "default"
+		}
+	}
+
+	cmd_postamble := ""
+	if gui {
+		mode_flag = " -gui"
+		if rule.WaveformInit != nil {
+			do_flags = append(do_flags, rule.WaveformInit.String())
+		}
+	} else {
+		if !print_output {
+			mode_flag = mode_flag + " -nostdout"
+		}
+		do_flags = append(do_flags, "\"run -all\"")
+		if Coverage.Value() {
+			do_flags = append(do_flags, fmt.Sprintf("\"coverage save -assert"+
+				" -cvg -codeAll -testname %s"+
+				" %s.ucdb\"",
+				testcase, coverage_db))
+			do_flags = append(do_flags,
+				fmt.Sprintf("\"vcover merge -testassociated -out %s.ucdb %s.ucdb %s.ucdb\"",
+					main_coverage_db, main_coverage_db, coverage_db))
+			if CoverageHtml.Value() {
+				do_flags = append(do_flags,
+					fmt.Sprintf("\"vcover report -html -output %s_covhtml -testdetails -details -assert"+
+						" -cvg -codeAll %s.ucdb\"", main_coverage_db, main_coverage_db))
+			}
+		}
+		do_flags = append(do_flags, "\"quit -code [coverage attribute -name TESTSTATUS -concise]\"")
+		cmd_newline := ":"
+		if cmd_echo != "" {
+			cmd_newline = "echo"
+		}
+		cmd_postamble = fmt.Sprintf("|| { %s; cat %s; exit 1; }", cmd_newline, log_file.String())
+	}
+
+	vsim_flags = vsim_flags + mode_flag + seed_flag + coverage_flag +
+		verbosity_flag + plusargs_flag + VsimFlags.Value()
+
+	for _, do_flag := range do_flags {
+		vsim_flags = vsim_flags + " -do " + do_flag
+	}
+
+	cmd := fmt.Sprintf("{ echo -n %s && vsim %s -work %s %s && echo PASS; }", cmd_echo, vsim_flags, rule.Lib(), rule.Target()+params)
```

Comment:
Also this doesn't work reliably with the current top-level tests.

Reply:
Should have been fixed now.